torchgeo/models/dofa.py

```diff
@@ -0,0 +1,544 @@
+
+        # timm's trunc_normal_(std=.02) is effectively normal_(std=0.02) as cutoff is
+        # too big (2.)
+        torch.nn.init.normal_(self.weight_tokens, std=0.02)
+        torch.nn.init.normal_(self.bias_token, std=0.02)
+
+    def forward(self, x: Tensor) -> tuple[Tensor, Tensor]:
+        """Forward pass of the model.
+
+        Args:
+            x: Input mini-batch of size (seq_len, batch, input_dim).
+
+        Returns:
+            Weight and bias.
+        """
+        pos_wave = x
+        x = torch.cat([self.weight_tokens, pos_wave], dim=0)
+        x = torch.cat([x, self.bias_token], dim=0)
+        transformer_output = self.transformer_encoder(x)
+        weights = self.fc_weight(transformer_output[self.wt_num : -1] + pos_wave)
+        # Using the last output to generate bias
+        bias = self.fc_bias(transformer_output[-1])
+        return weights, bias
+
+
+class FCResLayer(nn.Module):
+    """Fully-connected residual layer."""
+
+    def __init__(self, linear_size: int = 128) -> None:
+        """Initialize a new FCResLayer instance.
+
+        Args:
+            linear_size: Size of linear layer.
+        """
+        super().__init__()
+        self.l_size = linear_size
+        self.nonlin1 = nn.ReLU(inplace=True)
+        self.nonlin2 = nn.ReLU(inplace=True)
+        self.w1 = nn.Linear(self.l_size, self.l_size)
+        self.w2 = nn.Linear(self.l_size, self.l_size)
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Forward pass of the model.
+
+        Args:
+            x: Input mini-batch.
+
+        Returns:
+            Output of the model.
+        """
+        y = self.w1(x)
+        y = self.nonlin1(y)
+        y = self.w2(y)
+        y = self.nonlin2(y)
+        out: Tensor = x + y
+        return out
+
+
+class Dynamic_MLP_OFA(nn.Module):
+    """Dynamic Multilayer Perceptron for DOFA."""
+
+    def __init__(
+        self,
+        wv_planes: int,
+        inter_dim: int = 128,
+        kernel_size: int = 3,
+        embed_dim: int = 1024,
+    ) -> None:
+        """Initialize a new Dynamic_MLP_OFA instance.
+
+        Args:
+            wv_planes: Dimensions of dynamic weight generator.
+            inter_dim: Internal dimensions.
+            kernel_size: Kernel size of the depth-wise convolution.
+            embed_dim: Embedding dimensions.
+        """
+        super().__init__()
+        self.wv_planes = wv_planes
+        self.inter_dim = inter_dim
+        self.kernel_size = kernel_size
+        self.embed_dim = embed_dim
+        self._num_kernel = self.kernel_size * self.kernel_size * self.embed_dim
+        self.patch_size = (kernel_size, kernel_size)
+        self.num_patches = -1
+
+        self.weight_generator = TransformerWeightGenerator(
+            wv_planes, self._num_kernel, embed_dim
+        )
+        self.scaler = 0.01
+
+        self.fclayer = FCResLayer(wv_planes)
+
+        self._init_weights()
+
+    def _get_weights(self, waves: Tensor) -> Tensor:
```

Comment:
Probably going to get rid of this helper function, it doesn't do much and is only 1 line of code